blockchain/blockindex.go

```diff
@@ -49,6 +67,9 @@ type blockNode struct {
 	nonce      uint32
 	timestamp  int64
 	merkleRoot chainhash.Hash
+
+	// status is a bitfield representing the validation state of the block
+	status blockStatus
```

Comment:
With padding, this additional byte takes up a whole word. If we switch timestamp to be a uint32 (which is reasonable given that that is how it is serialized in the block message), we could avoid using the additional 8 bytes. However, that requires adding a bunch of casting in various parts of the code where timestamp is converted to/from a `time.Time`.

Reply:
I personally would prefer to just eat the extra 4 bytes.  I realize that currently it does get serialized to the wire as a uint32 (and is indeed part of the consensus rules because the hash is committing to that uint32 serialized value).  However, working with timestamps that aren't `int64` is not very natural in my opinion.  There are places in the code that do it, but it is much more natural to work with `int64` timestamps since that is really sort of the standard when dealing with unix-style timestamps.  It is also more natural to be able to subtract two timestamps and result in a negative time instead of worrying about overflowing.